internal/cluster/certificates.go

```diff
@@ -194,6 +195,22 @@ func (c Certificates) Lookup(ctx context.Context, ctrlclient client.Client, clus
 			return err
 		}
 		certificate.KeyPair = kp
+
+		// Ensure Secret is owned by the Cluster. This provides a migration path for Secrets
+		// previously owned by KubeadmConfigs.
+		for _, ownerRef := range s.OwnerReferences {
+			if ownerRef.Kind == "KubeadmConfig" {
```

Comment:
```suggestion
			if ownerRef.Kind != "Cluster" {
```

Reply:
@vincepri we just check to see if it is not owned by a Cluster and someone is managing the secret externally and setting their own OwnerRef on it, then we would take ownership of it erroneously.